#! /bin/sh /usr/share/dpatch/dpatch-run
## 005_fix-pthreads-detect.dpatch by  <divanov@creditreform.bg>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix pthreads detection

@DPATCH@

--- firebird2-1.5.2.orig/src/common/classes/locks.h
+++ firebird2-1.5.2/src/common/classes/locks.h
@@ -29,24 +29,22 @@
 #include "firebird.h"
 
 #ifdef MULTI_THREAD
-#ifdef WIN_NT
+#if defined(WIN_NT)
 // It is relatively easy to avoid using this header. Maybe do the same stuff like
 // in thd.h ? This is Windows platform maintainers choice
 #include <windows.h>
-#else
-#ifndef SOLARIS
-#include <pthread.h>
-#else
+#elif defined(SOLARIS)
 #include <thread.h>
 #include <synch.h>
-#endif
+#else
+#include <pthread.h>
 #endif
 #endif /* MULTI_THREAD */
 
 namespace Firebird {
 
 #ifdef MULTI_THREAD
-#ifdef WIN_NT
+#if defined(WIN_NT)
 
 /* Process-local spinlock. Used to manage memory heaps in threaded environment. */
 // Windows version of the class
@@ -77,7 +75,7 @@
 
 /* Process-local spinlock. Used to manage memory heaps in threaded environment. */
 // Pthreads version of the class
-#if !defined(SOLARIS) && !defined(DARWIN) && !defined(FREEBSD)
+#if !defined(SOLARIS) && !defined(DARWIN) && !defined(FREEBSD) && !defined(_POSIX_THREAD_IS_GNU_PTH)
 class Spinlock {
 private:
 	pthread_spinlock_t spinlock;
@@ -124,7 +122,7 @@
 			system_call_failed::raise();
 	}
 };
-#else  // DARWIN and FREEBSD
+#else  // DARWIN, FREEBSD and gnu pth
 class Spinlock {
 private:
 	pthread_mutex_t mlock;
